app/templates/components/orders/order-summary.hbs

```diff
@@ -39,63 +39,106 @@
     <h3 class="weight-400">{{t 'Order Summary'}}</h3>
   </div>
   {{#if (or (eq session.currentRouteName 'orders.new') (eq session.currentRouteName 'orders.view') (eq session.currentRouteName 'orders.pending'))}}
-    <table class="ui very basic tablet stackable table order-summary center aligned">
-      <thead>
-        <tr>
-          <th class="four wide">{{t 'Ticket Type'}}</th>
-          <th class="four wide">{{t 'Price'}}</th>
-          <th class="ui aligned two wide">{{t 'Discount'}}</th>
-          <th class="one wide">{{t 'Quantity'}}</th>
-          <th class="ui aligned two wide">{{t 'Subtotal'}}</th>
-          {{#if event.tax}}
-            <th class="four wide">{{t 'Tax'}}</th>
-          {{/if}}
-        </tr>
-      </thead>
-      <tbody>
-        {{#each tickets as |ticket|}}
+    {{#if (eq data.amount 0)}}
+      <table class="ui very basic tablet stackable table order-summary center aligned">
+        <thead>
           <tr>
-            <td>
-              <div class="ui small">
-                {{ticket.name}}
-              </div>
-            </td>
-            <td>{{currency-symbol eventCurrency}} {{format-number ticket.price}}</td>
-            <td>{{currency-symbol eventCurrency}} {{format-number ticket.discount}}</td>
-            <td>{{ticket-attendees data.attendees ticket.attendees}}</td>
-            <td>
-              {{currency-symbol eventCurrency}} {{format-number (mult (sub ticket.price ticket.discount) (ticket-attendees data.attendees ticket.attendees))}}
-            </td>
+            <th class="ui aligned two wide">{{t 'Ticket Type'}}</th>
+            {{#if data.discountCode}}
+              <th class="two wide">{{t 'Price'}}</th>
+              <th class=" two wide">{{t 'Discount'}}</th>
+            {{/if}}
+            <th class="two wide">{{t 'Quantity'}}</th>
+          </tr>
+        </thead>
+        <tbody>
+          {{#each tickets as |ticket|}}
+            <tr>
+              <td>
+                <div class="ui small">
+                  {{ticket.name}}
+                </div>
+              </td>
+              {{#if data.discountCode}}
+                <td>{{currency-symbol eventCurrency}} {{format-number ticket.price}}</td>
+                <td>{{currency-symbol eventCurrency}} {{format-number ticket.discount}}</td>
+              {{/if}}
+              <td>{{ticket-attendees data.attendees ticket.attendees}}</td>
+            </tr>
+          {{/each}}
+        </tbody>
+      </table>
+    {{else}}
+      <table class="ui very basic tablet stackable table order-summary center aligned">
```

Comment:
@kushthedude there is a lot of duplicate lines from here onwards which can be easily reduced by changing the condition checks and using it at appropriate places. Can we change it and reduce the duplication of code?

Reply:
@shreyanshdwivedi I checked the lines again , If you actually see the code, the one which seems duplicate are mere condition and , I have tried to use less condition as much as possible , After the following there are mere two checks one about `Ticket Price` and one about `Ticket Discount` , I can even merge the checks but this wont be of any good as number of lines will be same . 
If you can find any duplication , Please report I will reduce them asap . Thanks

Reply:
@kushthedude the lines 43-46 and lines 72-75 can be merged. The check on 47 and 77 can also be at one place. There are many such places which will reduce the duplication where check at line 42 can be used also.

Reply:
The check for 47 and 77 cant be at one place, As the one of 47 checks for 100 % discount as the net amount is zero and one at 77 checks for the if there is any discount and both will show different columns according to following

Reply:
@shreyanshdwivedi The one for 43-46 I really missed that Thanks for pointing